src/pages/Product-Listing/Filters/Filters.js

```diff
@@ -0,0 +1,124 @@
+import React from "react";
+import { useStateContext } from "../../../context/StateProvider";
+import "./filters.css";
+
+export const Filters = () => {
+  const {
+    productsList,
+    filterDispatch,
+    productTypesArray,
+    ratingInput,
+    sortByPrice,
+  } = useStateContext();
+
+  const productTypes = [
+    ...new Set(productsList.map((product) => product.productType)),
+  ];
+
+  const handleChange = (e) => {
+    let arr = [...productTypesArray];
+    if (e.target.checked) {
+      arr.push(e.target.value);
+    } else {
+      arr = arr.filter((productType) => productType !== e.target.value);
+    }
+    filterDispatch({ type: "SET_PRODUCT_TYPE", payload: arr });
+  };
+
+  return (
+    <aside className="filters">
+      <div className="underline flex-center">
+        <h5 className="filters-title">Filter by</h5>
+        <span
+          onClick={() => filterDispatch({ type: "CLEAR_FILTERS" })}
+          className="text-right"
+        >
+          Clear
+        </span>
+      </div>
+      <ul>
+        <li className=" filter-type">
+          <p className="filter-type-title">Collection </p>
+          <form className="sub-categories">
+            {productTypes.map((type, index) => {
+              return (
+                <label className="category" for={type} key={index}>
+                  <input
+                    value={type}
+                    checked={productTypesArray.some(
```

Comment:
appreciation - good use of `some()`

Reply:
I came across this while reviewing code of others'.